api/base/serializers.py

```diff
@@ -562,11 +575,24 @@ def to_representation(self, obj, envelope='data'):
         type_ = getattr(meta, 'type_', None)
         assert type_ is not None, 'Must define Meta.type_'
 
-        data = collections.OrderedDict([('id', ''), ('type', type_), ('attributes', collections.OrderedDict()),
-                                        ('relationships', collections.OrderedDict()), ('links', {})])
+        data = collections.OrderedDict([
+            ('id', ''),
+            ('type', type_),
+            ('attributes', collections.OrderedDict()),
+            ('relationships', collections.OrderedDict()),
+            ('embeds', {}),
+            ('links', {}),
+        ])
 
+        embeds = self.context.get('embed', {})
         fields = [field for field in self.fields.values() if not field.write_only]
 
+        for item in set(embeds.keys()) - set([f.field_name for f in fields if getattr(f, 'json_api_link', False)]):
```

Comment:
The `for` loop seems a bit awkward, since there can only ever be one iteration here. Since you're computing all the invalid embed field names, why not just pass them all in the error message?

```python
invalid_embeds = set(embeds.keys()) - set([f.field_name for f in fields if getattr(f, 'json_api_link', False)])
if invalid_embeds:
    raise InvalidQueryStringError(..., detail='The following fields are not embeddable: {}'.format(', '.join(invalid_embeds))
```